djangocms_navigation/static/djangocms_navigation/js/navigation-tree-admin.js

```diff
@@ -0,0 +1,367 @@
+                                            // if not create element to append message(s) too:
+                                            let msglist = document.getElementsByClassName('messagelist')[0];
+                                            if(msglist === undefined){
+                                                msglist = document.createElement('ul');
+                                                msglist.className = 'messagelist';    
+                                                // Insert messagelist before main content div:
+                                                document.getElementById('content').before(msglist);
+                                            }
+                                            
+                                            payload.messages.forEach(message => {
+                                                let msg = document.createElement('li');
+                                                msg.className = message.level;
+                                                msg.innerHTML = message.message;
+
+                                                msglist.append(msg);
+                                            })
+                                        })
+                                        .catch(err => {
+                                            console.log(`Error: Unfortunately there was an error: ${err}`)
+                                        });
+                                    }
+                                });
+                            }
+                        }
+                        stop_drag();
+                    }).bind('keyup', function (kbevt) {
+                        // Cancel drag on escape
+                        if (kbevt.keyCode === 27) {
+                            stop_drag();
+                        }
+                    });
+            });
+    
+            $('a.collapse').click(function () {
+                var node = new Node($(this).closest('tr')[0]); // send the DOM node, not jQ
+                node.toggle();
+                return false;
+            });
+            var hash = window.location.hash;
+            // This is a hack, the actual element's id ends in '-id' but the url's hash
+            // doesn't, I'm doing this to avoid scrolling the page... is that a good thing?
+            if (hash) {
+                $(hash + '-id').animate({
+                    backgroundColor: RECENTLY_MOVED_COLOR
+                }, RECENTLY_FADE_DURATION, function () {
+                    $(this).animate({
+                        backgroundColor: RECENTLY_MOVED_FADEOUT
+                    }, RECENTLY_FADE_DURATION, function () {
+                        this.removeAttribute('style');
+                    });
+                });
+            }
+        });
+    })(django.jQuery);
+    
+    // http://stackoverflow.com/questions/190560/jquery-animate-backgroundcolor/2302005#2302005
```

Comment:
We should add a why this block is here, if one day this is a library or built in we would want to know if it can be removed / replaced.